frontend/src/metabase/components/MetadataInfo/DimensionInfoPopover/DimensionInfoPopover.tsx

```diff
@@ -20,13 +21,24 @@ type Props = { dimension: Dimension } & Pick<
   "children" | "placement"
 >;
 
+function checkForMetadata(dimension: Dimension): boolean {
+  const query = dimension?.query?.();
+  if (dimension && query) {
+    return query.isEditable();
```

Comment:
`isEditable` actually calls the `hasMetadata` method. It might be a bit more reasonable to use `hasMetadata` here for readability

Reply:
good call, done, except I used `query.metadata()` because `hasMetadata` isn't implemented on native queries